nodeadm/internal/kubelet/daemon.go

```diff
@@ -44,8 +49,20 @@ func (k *kubelet) Configure(cfg *api.NodeConfig) error {
 	return nil
 }
 
-func (k *kubelet) EnsureRunning() error {
-	return k.daemonManager.StartDaemon(KubeletDaemonName)
+func (k *kubelet) EnsureRunning(ctx context.Context) error {
+	if err := k.daemonManager.StartDaemon(KubeletDaemonName); err != nil {
+		return err
+	}
+	return util.NewRetrier(util.WithRetryAlways(), util.WithBackoffFixed(250*time.Millisecond)).Retry(ctx, func() error {
+		status, err := k.daemonManager.GetDaemonStatus(KubeletDaemonName)
+		if err != nil {
+			return err
+		}
+		if status != daemon.DaemonStatusRunning {
+			return fmt.Errorf("%s status is not %q", KubeletDaemonName, daemon.DaemonStatusRunning)
+		}
+		return nil
+	})
```

Comment:
this block is identical in `containerd/daemon.go`, any reason we can't move this down into daemonmanager.StartDaemon?

Reply:
im good with that if it means we change the expectation of `StartDaemon` to essentially be the same as `EnsureRunning`. ie. block until running/active

Reply:
for better uniformity i pushed the waiter logic into the `daemonmanager` and enforced it in `Start`, `Stop`, and `Restart`